### PR TITLE
Skip unnecessary calls if user is not logged in

### PR DESCRIPTION
**Pull request by joshenlim**

### CODE REVIEW

File: studio/components/interfaces/App/RouteValidationWrapper.tsx [hunk 1]
```diff
@@ -16,6 +16,9 @@ const RouteValidationWrapper = ({ children }: PropsWithChildren<{}>) => {
   const { ref, slug } = useParams()
   const navLayoutV2 = useFlag('navigationLayoutV2')
 
+  const { session } = useAuth()
+  const isLoggedIn = Boolean(session)
```
Comment (alaister) [inline studio/components/interfaces/App/RouteValidationWrapper.tsx:20]: You can use `useIsLoggedIn()` from `lib/auth` here :)

**Review by alaister — COMMENTED** [review]
Looks good! There's a handy hook already for logged in status:

File: studio/components/interfaces/App/RouteValidationWrapper.tsx (hunk 1 above)
Comment (joshenlim) [reply studio/components/interfaces/App/RouteValidationWrapper.tsx:20]: oooh okay gotcha - i'll update accordingly 🙏

Comment (joshenlim) [reply studio/components/interfaces/App/RouteValidationWrapper.tsx]: just pushed :)

File: studio/components/ui/PageTelemetry.tsx [hunk 2]
```diff
@@ -59,14 +61,16 @@ const PageTelemetry = ({ children }: PropsWithChildren<{}>) => {
...
 
-      post(`${API_URL}/telemetry/pageview`, {
-        ...(ref && { projectRef: ref }),
-        ...(selectedOrganization && { orgSlug: selectedOrganization.slug }),
-        referrer: referrer,
-        title: document.title,
-        path: router.route,
-        location: router.asPath,
-      })
+      if (isLoggedIn) {
```
Comment (alaister) [inline studio/components/ui/PageTelemetry.tsx:64]: Just to confirm, we only want to register page views for logged in users? (aka we don't care about the views on the sign in pages?)

Comment (joshenlim) [reply studio/components/ui/PageTelemetry.tsx:64]: @pamelachia - best to check with you about this. the only reason im guarding the pageview call is cause it currently 401s on the sign in pages

Comment (alaister) [reply studio/components/ui/PageTelemetry.tsx:64]: oh if it's 401 already, I assume it's fine!

**Review by alaister — APPROVED** [review]
LGTM!

**Pull request merged**